### ISSUE TITLE
AssertError when opening preferences

### ISSUE THREAD

**Issue by alexwhitman** [gnunn1/tilix#1166]
Trying to build/run latest master on Ubuntu 17.10.  I applied the patch used in the Ubuntu package so that it will work without PCRE2:

```diff
diff --git a/source/gx/tilix/terminal/regex.d b/source/gx/tilix/terminal/regex.d
index fc42834..3323a8a 100644
--- a/source/gx/tilix/terminal/regex.d
+++ b/source/gx/tilix/terminal/regex.d
@@ -234,7 +234,7 @@ VRegex compileVRegex(TerminalRegex regex) {
 static this() {
     import std.exception : assumeUnique;
 
-    if (checkVTEVersionNumber(0, 46)) {
+    if (checkVTEVersionNumber(0, 60)) {
         VRegex[URL_REGEX_PATTERNS.length] tempRegex;
         foreach (i, regex; URL_REGEX_PATTERNS) {
             tempRegex[i] = compileVRegex(regex);
diff --git a/source/gx/tilix/terminal/search.d b/source/gx/tilix/terminal/search.d
index 2eb58c7..a8cc39b 100644
--- a/source/gx/tilix/terminal/search.d
+++ b/source/gx/tilix/terminal/search.d
@@ -199,7 +199,7 @@ private:
     void setTerminalSearchCriteria() {
         string text = seSearch.getText();
         if (text.length == 0) {
-            if (checkVTEVersionNumber(0, 46)) {
+            if (checkVTEVersionNumber(0, 60)) {
                 vte.searchSetRegex(null, 0);
             } else {
                 vte.searchSetGregex(null, cast(GRegexMatchFlags) 0);
@@ -212,7 +212,7 @@ private:
             text = format("\\b%s\\b", text);
 
         try {
-            if (checkVTEVersionNumber(0, 46)) {
+            if (checkVTEVersionNumber(0, 60)) {
                 uint flags = PCRE2Flags.UTF | PCRE2Flags.MULTILINE | PCRE2Flags.NO_UTF_CHECK;
                 if (!matchCase) {
                     flags |= PCRE2Flags.CASELESS;
diff --git a/source/gx/tilix/terminal/terminal.d b/source/gx/tilix/terminal/terminal.d
index f948512..70b62ee 100644
--- a/source/gx/tilix/terminal/terminal.d
+++ b/source/gx/tilix/terminal/terminal.d
@@ -863,13 +863,13 @@ private:
         vte.setVexpand(true);
         //Search Properties
         vte.searchSetWrapAround(gsSettings.getValue(SETTINGS_SEARCH_DEFAULT_WRAP_AROUND).getBoolean());
-        if (checkVTEVersionNumber(0, 49)) {
+        if (checkVTEVersionNumber(0, 60)) {
             vte.setAllowHyperlink(true);
-            trace("Custom hyperlinks enabled for VTE 0.49");
+            trace("Custom hyperlinks enabled for VTE 0.50");
         }
         //URL Regex Experessions
         try {
-            if (checkVTEVersionNumber(0, 46)) {
+            if (checkVTEVersionNumber(0, 60)) {
                 foreach (i, regex; compiledVRegex) {
                     int id = vte.matchAddRegex(cast(VRegex) regex, 0);
                     regexTag[id] = URL_REGEX_PATTERNS[i];
@@ -1705,7 +1705,7 @@ private:
     }
 
     public void checkHyperlinkMatch(Event event) {
-        if (!checkVTEVersionNumber(0, 49)) return;
+        if (!checkVTEVersionNumber(0, 60)) return;
         string uri = vte.hyperlinkCheckEvent(event);
         if (uri.length == 0) return;
         match.match = uri;
@@ -2368,7 +2368,7 @@ private:
                 }
                 TerminalRegex regex = TerminalRegex(value[0], TerminalURLFlavor.CUSTOM, caseInsensitive, value[1]);
                 try {
-                    if (checkVTEVersionNumber(0, 46)) {
+                    if (checkVTEVersionNumber(0, 60)) {
                         VRegex compiledRegex = compileVRegex(regex);
                         if (compiledRegex !is null) {
                             int id = vte.matchAddRegex(compiledRegex, 0);
```

Applying the patch worked fine, tilix builds and runs but trying to open the preferences dialog gives:
```
core.exception.AssertError@../../.dub/packages/gtk-d-3.7.0/gtk-d/generated/gtkd/gobject/ObjectG.d(159): null this
----------------
??:? _d_assert_msg [0xe38b6e1a]
??:? pure nothrow @nogc @safe gx.tilix.prefeditor.prefdialog.ProfilePreferenceRow gobject.ObjectG.ObjectG.opCast!(gx.tilix.prefeditor.prefdialog.ProfilePreferenceRow).opCast() [0xe350af61]
??:? void gx.tilix.prefeditor.prefdialog.PreferenceDialog.updateUI() [0xe3589a18]
??:? gx.tilix.prefeditor.prefdialog.PreferenceDialog gx.tilix.prefeditor.prefdialog.PreferenceDialog.__ctor(gtk.ApplicationWindow.ApplicationWindow) [0xe358a246]
??:? void gx.tilix.application.Tilix.presentPreferences() [0xe3534a47]
??:? void gx.tilix.application.Tilix.onShowPreferences() [0xe3531845]
??:? void gx.tilix.application.Tilix.installAppMenu().__dgliteral5!(glib.Variant.Variant, gio.SimpleAction.SimpleAction).__dgliteral5(glib.Variant.Variant, gio.SimpleAction.SimpleAction) [0xe353163b]
??:? extern (C) void gio.SimpleAction.SimpleAction.callBackActivate(gio.c.types.GSimpleAction*, glib.c.types.GVariant*, gio.SimpleAction.SimpleAction.OnActivateDelegateWrapper) [0xe363ce39]
??:? g_closure_invoke [0x39c8df9c]
```

**Comment by gnunn1** [comment]
Use ldc2 to compile for now, there's an issue with dmd that is being looked at by the GtkD maintainer.

**Comment by alexwhitman** [comment]
Ah, fair enough.  For now I've just exported/imported settings from another machine and tilix appears to be working fine as long as I don't open the preferences dialog.

**Comment by gnunn1** [comment]
References here:

https://forum.gtkd.org/groups/GtkD/thread/661/

and here:

http://forum.dlang.org/thread/ot20f2$2gub$1@digitalmars.com

**Comment by alexwhitman** [comment]
So based off a comment in one of the threads above, I updated DMD to beta 2.077.0-rc.1 and now the preference dialog works.  There were a bunch of deprecation warnings in the build but that's not a problem:
```
source/gx/gtk/cairo.d(40,15): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/cairo.d(40,15): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/cairo.d(40,20): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/cairo.d(40,30): Deprecation: alias std.datetime.AutoStart is deprecated - Use std.datetime.stopwatch.AutoStart.
source/gx/gtk/cairo.d(134,15): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/cairo.d(134,15): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/cairo.d(134,20): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/cairo.d(134,30): Deprecation: alias std.datetime.AutoStart is deprecated - Use std.datetime.stopwatch.AutoStart.
source/gx/gtk/util.d(71,12): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/util.d(71,12): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
source/gx/gtk/util.d(71,12): Deprecation: alias std.datetime.AutoStart is deprecated - Use std.datetime.stopwatch.AutoStart.
source/gx/gtk/util.d(71,12): Deprecation: alias std.datetime.AutoStart is deprecated - Use std.datetime.stopwatch.AutoStart.
source/gx/gtk/util.d(72,15): Deprecation: struct std.datetime.StopWatch is deprecated - Use std.datetime.stopwatch.StopWatch.
```

**Comment by gnunn1** [comment]
This has been fixed in Gtkd 1.7.1, I have updated dub.json to use that version now.
